tox.ini

```diff
@@ -11,7 +11,12 @@ deps =
 
 commands =
     flake8
-    py.test --cov=shub --cov-report= {posargs:shub tests}
+    py.test -p no:warnings --cov=shub --cov-report= {posargs:shub tests}
```

Comment:
@rafaelcapucho  Alright, the CI tests pass now (almost all, a few ones still fail in AppVeyor), but to make it work I had to completely disable warnings: there's some issue happening on python>3.5 that indefinitely retries a test when a warning is yielded, it took me a while before I realized what's happening. While it works, I believe we need to address it before merging the changes. It doesn't happen on py3.5 with exactly the same dependencies, so probably something related to some pytest setting (or any related plugins/dependencies) working differently on py>3.5.

It's reproducible locally  (python 3.6, `pip install -r requirement.txt -r tests/requirements.txt, py.test -v`)
The test I used to reproduce it is this one `test_config.py::LoadShubConfigTest::test_no_global_scrapinghub_yml`, it hangs forever when running locally (and gradually eats all memory available) because of the warnings. If you could take a look and identify/fix the reason, it'd be awesome.

Reply:
@vshlapakov Hi Vik, The problem is the following, In the version `pytest>=3.1.0` they changed the way they handle warnings:

![image](https://user-images.githubusercontent.com/163367/86715930-164a0b00-bff7-11ea-8636-ff4e72993bee.png)

And they said that it might fail when the user's code tries to treat the warning, instead of using the standard, it is what we are doing in `shub/exceptions.py`:

```
def print_warning(msg, category=ShubWarning):
    """Helper to use Python warnings with custom formatter."""

    def custom_showwarning(message, *args, **kwargs):
        # ignore everything except the message
        try:
            sys.stderr.write("WARNING: " + str(message) + '\n')
        # stderr is invalid - this warning just gets lost
        except (IOError, UnicodeError):
            pass

    old_showwarning = warnings.showwarning
    try:
        warnings.showwarning = custom_showwarning
        warnings.warn(msg, category=category)
    finally:
        warnings.formatwarning = old_showwarning
```

Here we are overwriting the `.showwarning` attribute, I did a quick test by allowing the warnings to return again and with:

```
def print_warning(msg, category=ShubWarning):
    warnings.warn(msg, category=category)
```
and the tests started to work again: [qa1wsi9j5siafs4t](https://ci.appveyor.com/project/scrapinghub/shub/builds/33946544/job/qa1wsi9j5siafs4t)

`.showwarning` attribute is used when writing the warning to a file ([Docs](https://docs.python.org/3.8/library/warnings.html#warnings.showwarning)). 

I'm going to think about an approach to sort it.

Reply:
Amazing, thanks :100: